packages/core/src/node/messaging/ipc-protocol.ts

```diff
@@ -8,7 +8,41 @@
 
 import { MessageConnection } from "vscode-jsonrpc";
 
-export const THEIA_PARENT_PID = 'THEIA_PARENT_PID';
-export const THEIA_ENTRY_POINT = 'THEIA_ENTRY_POINT';
+const THEIA_PARENT_PID = 'THEIA_PARENT_PID';
+const THEIA_ENTRY_POINT = 'THEIA_ENTRY_POINT';
 
 export type IPCEntryPoint = (connection: MessageConnection) => void;
+
+/**
+ * Exit the current process if the parent process is not alive.
+ * Relevant only for some OS, like Windows
+ */
+export function checkParentAlive(): void {
+    if (process.env[THEIA_PARENT_PID]) {
+        const parentPid = Number(process.env[THEIA_PARENT_PID]);
+
+        if (typeof parentPid === 'number' && !isNaN(parentPid)) {
+            setInterval(() => {
+                try {
+                    // throws an exception if the main process doesn't exist anymore.
```

Comment:
Nice one but, please please, add some more comment: `This method will throw an error if the target pid does not exist. As a special case, a signal of 0 can be used to test for the existence of a process.`

Reply:
Actually, as you feel. It was not obvious to me what it does.